src/Changes.php

```diff
@@ -0,0 +1,46 @@
+<?php
+declare(strict_types=1);
+
+namespace Roave\ApiCompare;
+
+use ArrayIterator;
+use Assert\Assert;
+use IteratorAggregate;
+
+final class Changes implements IteratorAggregate
+{
+    /** @var Change[] */
+    private $changes = [];
+
+    private function __construct()
```

Comment:
What is the purpose of a private constructor in this case?

Reply:
Named constructors etc.